### PR TITLE
Global timeline doesn't show total population correctly

### LINKED ISSUE #3611: Global timeline doesn't show total population correctly
There's probably some logic difference between the local and global timeline that causes the global timeline to have incorrect numbers. See issue reported here: https://github.com/Revolutionary-Games/Thrive/pull/3605#pullrequestreview-1052801452

### PR DESCRIPTION
**Pull request by Kashnox**
**Brief Description of What This PR Does**

This PR starts calling ApplyExternalEffects from within the CalculateFinalExternalEffectSizes method. The external effects still aren't actually applied to the world until ApplyAllResultsAndEffects is called, but they're stored in the AutoEvoRun.Results' patch information so summaries and timeline events can use them.

Previously, everything logging the results was manually adding the external effects except the global timeline.

**Related Issues**

closes #3611 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by Kashnox** [comment]
I tried to thoroughly test the Auto-Evo Prediction, the Auto-Evo summary in the editor, the Local and Global Timelines, and the Auto-Evo Exploring Tool to make sure they're still working correctly, but auto-evo is complicated, so someone who has more experience working with it should take a good look at my changes too.

**Comment by buckly90** [comment]
Numbers all seem to match up accordingly. No issues appearant. Well done!

File: src/auto-evo/AutoEvoRun.cs [hunk 1]
```diff
@@ -242,7 +242,7 @@ public bool IsFinished(bool autoStart = true)
     /// <remarks>
     ///   <para>
     ///     This has to be called after this run is finished.
-    ///     <see cref="CalculateFinalExternalEffectSizes"/> must be called first,
+    ///     <see cref="CalculateAndApplyFinalExternalEffectSizes"/> must be called first,
     ///     that should be called even before generating the result summaries to make sure they are accurate.
     ///   </para>
     /// </remarks>
```
Comment (hhyyrylainen) [inline src/auto-evo/AutoEvoRun.cs:248]: Just one small comment, should `ApplyAllResultsAndEffects` now be renamed to `ApplyAllResults`? As it no longer handles applying the external effects.

Comment (hhyyrylainen) [reply src/auto-evo/AutoEvoRun.cs:248]: The summary on this method is `Applies things added by addExternalPopulationEffect and auto-evo results` but I think it should now be `Applies computed auto-evo results to the world` as with the name this method is no longer responsible for handling the external effects instead the computed results need to be the final numbers to apply.

Comment (Kashnox) [reply src/auto-evo/AutoEvoRun.cs:248]: Ah, okay, that makes sense.

**Review by hhyyrylainen — APPROVED** [review]
This now makes sense to me. I'll trust that Buckly's testing was enough so I won't do a second playtest on this.

**Pull request merged**